src/utils/identifiers/three-words/index.js

```diff
@@ -0,0 +1,32 @@
+/*
+ * ISC License (ISC)
+ * Copyright (c) 2018 aeternity developers
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ *  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ *  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ *  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ *  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+
+import adjectives from './adjectives'
+import animals from './animals'
+import { decodeBase58Check } from '../../crypto'
+
+export default address => {
+  const bytes = decodeBase58Check(address.split('$')[1])
+  let entropy = bytes.readUInt32BE(bytes.length - 4)
```

Comment:
What if this function receives two different addresses with the same entropy? Looks like it would return the same words. Maybe each word should be based on separate entropy (i.e. different bytes of the buffer)?

Reply:
Yes, this function will return the same words if addresses have the same 4 last bytes, but it should be quite difficult for the attacker to generate a key pair with the same last bytes.

Also, this implementation does not protect against typos. Maybe aeternity accounts addresses are already somehow protected against typos?

Entropy can be generated based on the hash of an address or we can use account address as randseed like [Status does](https://github.com/status-im/status-react/blob/dc23ce28039f45204f71da609805fdb7d58f7136/src/status_im/utils/gfycat/core.cljs#L14), but as I understand the generation of such identifier is much cheaper than a generation of key pair so even if we will use the whole address as entropy still will be possible to generate different address with the same identifier, but without corresponding private key. Of course, we can increase the complexity of identifier generation algorithm, but it can decrease the performance of our apps in some cases.